server/server.go

```diff
@@ -256,17 +256,10 @@ func (s *Server) Run() error {
 		if s.shouldStopListener() {
 			err = conn.Close()
 			terror.Log(errors.Trace(err))
-			break
+			return errors.Trace(err)
 		}
 		go s.onConn(conn)
 	}
-	err := s.listener.Close()
```

Comment:
Why remove this?

Reply:
I don't like the sleep 1 minute loop here.
it will make server into a status that **it doesn't provide service any more but still alive**.
`cleanup()` in should do the graceful shut down.